### PR TITLE
v1.0.0

### PR DESCRIPTION
**Pull request by donavanbecker**
### Breaking Changes

- **The minimum Node.js version required is now `v18`.**
- **Important notice:** Because of the cleanup of the Deprecated code, you will need to migrate you code base.
    - Remove the long-deprecated init().
    - Deprecate Core, BridgedCore, legacy Camera characteristics. (#1058) (@hjdhjd)
        - For deprecated `Core` and `BridgedCore` see: https://github.com/homebridge/HAP-NodeJS/wiki/Deprecation-of-Core-and-BridgeCore
    - Legacy code deprecation cleanup. (#1059) (@hjdhjd)
        - For deprecated `storagePath` switch to `HAPStorage.setCustomStoragePath`, `AudioCodec` switch to `AudioStreamingCodec`, `VideoCodec` switch to `H264CodecParameters`,`StreamAudioParams` switch to `AudioStreamingOptions`, `StreamVideoParams` switch to `VideoStreamingOptions`,`cameraSource` switch to `CameraController`.
    - Others deprecated code to highlight removed: `useLegacyAdvertiser`, `AccessoryLoader`.
- Fix: Naming for Characteristic.ProgramMode has been corrected from `PROGRAM_SCHEDULED_MANUAL_MODE_` to `PROGRAM_SCHEDULED_MANUAL_MODE`

### Fixed

- Fix: Build Issues (#1041) (@NorthernMan54)
- Fix: Ensure data is only transmitted on open and ready connections. (#1051) (@hjdhjd)
- Fix: Ensure we check names using the full UTF-8 character set. (#1052) (@hjdhjd)
- Fix: ConfiguredName (#1049) (@donavanbecker)
- Fix: Manufacturer looking at checkName but should look at checkValue. (#1053) (@donavanbecker)

### Other Changes

- Implement warning messages for invalid characters in names (#1009) (@NorthernMan54)
- Mitigate event emitter "memory leak" warnings when a significant number of camera streaming events occur simultaneously (#1037) (@hjdhjd)
- AdaptiveLightingController fix & improvement (#1038) (@Shaquu)
- Minor fixes to recording logging and one change in logging. (#1040) (@hjdhjd)
- Bridged core and core cleanup (#1048) (@Shaquu)
- Increase snapshot handler warning timeout to 8000ms. (#1055) (@hjdhjd)
- Cleanup and refactor getLocalNetworkInterface and address a potential edge case. (#1056) (@hjdhjd)
- Correct log spacing
- Updated and fixed `typedoc` config file
- Updated dependencies

### Homebridge Dependencies

- `@homebridge/ciao` @ `v1.3.0`
- `bonjour-hap` @ `v3.8.0`

### CODE REVIEW

**Comment by Shaquu** [comment]
Wait for my change guys! 😅

**Comment by donavanbecker** [comment]
> Wait for my change guys! 😅

will do.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,13 +2,21 @@
 
 All notable changes to `hap-nodejs` will be documented in this file. This project tries to adhere to [Semantic Versioning](http://semver.org/).
 
-## v0.12.3 (2024-06-22)
+## v0.12.3 (2024-06-24)
```
Comment (Shaquu) [inline CHANGELOG.md]: Date to be bumped :)

**Review by Shaquu — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (bwp91) [reply CHANGELOG.md]: this line should be `BETA` until we release

File: package.json [hunk 2]
```diff
@@ -54,7 +54,7 @@
   "dependencies": {
```
Comment (Shaquu) [inline package.json:54]: Would be nice to bump package `version` already

Comment (bwp91) [reply package.json:54]: which?

Comment (Shaquu) [reply package.json:54]: Package, 0.12.2 to 0.12.3 or 0.12.3-whatever beta?

Comment (bwp91) [reply package.json:54]: this is set by the github actions to appropriate version if a beta or a release is being made

Comment (bwp91) [reply package.json:54]: if i haven’t answered your query please let me know!

**Comment by Shaquu** [comment]
May I ask what is the plan with 0.12.3? Anything I could help with to boost this release?

**Review by hjdhjd — DISMISSED** [review]

**Review by hjdhjd — APPROVED** [review]

**Pull request merged**